lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskQueryImpl.java

```diff
@@ -1643,6 +1679,163 @@ public TaskQuery orderByCustomAttribute(
     return addOrderCriteria(customField.name(), sortDirection);
   }
 
+  @Override
+  public TaskQuery customIntAttributeNotIn(TaskCustomIntField customIntField, Integer... values)
+      throws InvalidArgumentException {
+    if (values.length == 0) {
+      throw new InvalidArgumentException(
+          "At least one Integer has to be provided as a search parameter");
+    }
+    List<Integer> conditionList = new ArrayList<>(Arrays.asList(values));
+    switch (customIntField) {
+      case CUSTOM_INT_1:
+        this.customInt1NotIn = conditionList.toArray(new Integer[0]);
+        break;
+      case CUSTOM_INT_2:
+        this.customInt2NotIn = conditionList.toArray(new Integer[0]);
+        break;
+      case CUSTOM_INT_3:
+        this.customInt3NotIn = conditionList.toArray(new Integer[0]);
+        break;
+      case CUSTOM_INT_4:
+        this.customInt4NotIn = conditionList.toArray(new Integer[0]);
+        break;
+      case CUSTOM_INT_5:
+        this.customInt5NotIn = conditionList.toArray(new Integer[0]);
+        break;
+      case CUSTOM_INT_6:
+        this.customInt6NotIn = conditionList.toArray(new Integer[0]);
+        break;
+      case CUSTOM_INT_7:
+        this.customInt7NotIn = conditionList.toArray(new Integer[0]);
+        break;
+      case CUSTOM_INT_8:
+        this.customInt8NotIn = conditionList.toArray(new Integer[0]);
```

Comment:
why convert to list and then back to array?

Reply:
Because of the previous implementation of null handling. I will delete it